cmd/user/internal/domain/user/user.go

```diff
@@ -203,5 +203,8 @@ func (u *User) transition(e domain.RawEvent) {
 		u.email = e.Email
 	case EmailAddressWasChanged:
 		u.email = e.Email
+	case AccessTokenWasRequested:
+		u.id = e.ID
```

Comment:
ID of a user should not change, we want to set id on the aggregate root once when user is registered and then we should never override it.

event holds id to assign new email to already existing user

Reply:
Actually this should not be here, this event does not alter the state, in requests new auth token

the issue is somewhere else maybe in here ? https://github.com/vardius/go-api-boilerplate/blob/013ce81a35fed41a88a4c904dc842027824a7530/cmd/user/internal/domain/user/user.go#L40
when loading events from history ?

Reply:
How do I reproduce this issue to test ? What is the flow you perform ?

Reply:
I register a user and then request an access token. The logs I see are:

```
user_logs:

2019/12/31 20:44:40.629628 DEBUG: [EventBus|Publish]: user.AccessTokenWasRequested {"id":"d53836de-23d2-45a6-abb0-87f5abea8a2f","metadata":{"type":"user.AccessTokenWasRequested","stream_id":"00000000-0000-0000-0000-000000000000","stream_name":"user.User","stream_version":0,"occurred_at":"2019-12-31T20:44:40.628333849Z"},"payload":{"id":"00000000-0000-0000-0000-000000000000","email":""}}
2019/12/31 20:44:40.645071 INFO: [Request|End]: 7d0d9965-6268-45d7-9ceb-79e98b284ee6 : (0) : POST /v1/dispatch/request-user-access-token -> 172.17.0.8:45394 (23.73608ms)
2019/12/31 20:44:40.646782 DEBUG: [EventBus|Subscribe]: user.AccessTokenWasRequested {"id":"00000000-0000-0000-0000-000000000000","email":""}
2019/12/31 20:44:40 [EventHandler] {"id":"00000000-0000-0000-0000-000000000000","email":""}
2019/12/31 20:44:40 [EventHandler] Error: oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"invalid_request","error_description":"The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed"}

auth_logs:

2019/12/31 20:57:15.329019 ERROR: oAuth2 Server response error: invalid_client
&{invalid_client 0 Client authentication failed  401 map[]}
2019/12/31 20:57:15.331306 INFO: [Request|End]: 4a8f5593-ba15-444e-844e-6a7c381bedd1 : (0) : POST /v1/token -> 172.17.0.15:56618 (5.698101ms)
2019/12/31 20:57:15.332723 INFO: [Request|Start]: 490cbe9b-dd4c-43de-92eb-5abbb59cea66 : (0) : POST /v1/token -> 172.17.0.15:56618
2019/12/31 20:57:15.333062 ERROR: oAuth2 Server response error: invalid_request
&{invalid_request 0 The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed  400 map[]}
```